### DISCUSSION TITLE
ThinkSmart View - crash to logo when app set to auto start

### DISCUSSION THREAD

**Discussion by pgale** [dinki/View-Assist#12] [General]
On the TS View on A11 go apps ROM, has anyone managed to fix the crash to Lenovo logo when any app is set to auto start? As mentioned in the docs. Seems that it might be a bit more than just an app set to auto start as this happened to me after enabling root options in the Fully Kiosk browser settings. Might be coincidence of course.

Understand it might be down to the network (Wi-Fi) drivers or something else related. Is there any hope of getting a resolution for that?

Thanks.

**Comment by dinki** [comment]
I'm not certain that this is what really is happening.  The portion of the wiki I wrote is some of the earliest stuff.   I am not convinced that this is not the wifi issue just somewhat delayed.  

Are you set to not autoconnect wifi on boot?  If not, can you try, leave auto start apps on and see if you have the same issue?  Further, to be safe for your test, try unplugging the device for ten minutes and then boot and see if the above suggestion still results in a crash.   I've wanted to test this for a while but I've got a lot of other things ahead of it.

**Reply by pgale** [reply]
I'm occasionally seeing the crash as I install new things/apps. I can confirm that I CAN boot if I block my device on my Uni-Fi AP, let it boot, then unblock it. I have disabled autoconnect wifi on boot and I'm pretty sure it still crashes. I'll do yet more digging though. Thanks.

I'm just experimenting with the app "Automate" (free) which allows you to setup automated flows. I've created a flow that connects to wifi and starts Fully Kiosk app - but only part way through testing that. It needs loads of permissions setting so is relatively complex or time consuming to do on multiple devices (when I eventually convert the other four TS View's to A11). I'll post back if I get anywhere with it.

**Reply by dinki** [reply]
We're thinking alike.  I have used Automate in the past and was planning to get that going for the wifi reenable,  ensure that RTPMic and Wallpanel stay running.  Again, one of those things where I get to a point where I have time or need to try so if you can share your findings it will certainly help me and probably others.    

I may start a TS channel on the Discord server to bounce ideas and experiences on.

**Reply by pgale** [reply]
I've been trying to get Automate working to connect to a wifi network but it isn't working. Apparently this functionality was removed in Android 10. There is a legacy extension for Automate that helps with some of these issues but it didn't help connecting to a network and still doesn't work. Turning the wifi on/off does work though so I was wondering how to default to the wifi being off on boot but don't know how to do that or if it's possible.

**Reply by dinki** [reply]
Yeah nothing is easy is it?  My hope was to do what you're trying with the wifi connect.  The only other option I can think of to try is using Tasker but it's a bit more complicated in use if I remember correctly.

**Reply by pgale** [reply]
I think Tasker has the same issues from a quick search of WiFi issues - but I've not tried it. Wonder if deadman has any idea on the route issue although dev on A11 etc seems to have stopped.

**Reply by dinki** [reply]
I think I'm going to try PMOS when my new TS arrives.

**Reply by pgale** [reply]
I did run it for a while (about 3-4 weeks ago), but there were loads of issues with the display, flickering and scaling. Tiny on screen keyboard for me too. 
The biggest issue though is no Fully Kiosk which I've come to rely on and built management tools around. If only they did a Linux version.

**Reply by dinki** [reply]
Ugh..  That's not overly encouraging.  A guy on Discord was showing me some videos that made it look like a nice option.  I have no issue with switching to a 'regular' browser versus Wallpanel.  This discussion might be better to have on Discord! ha
